org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/backend/SonarLintEclipseHeadlessRpcClient.java

```diff
@@ -102,7 +105,12 @@ public List<ClientFileDto> listFiles(String configScopeId) throws ConfigScopeNot
     // For root project in root projects add the files to "files"
     for (var rootProject : rootProjects) {
       FileSystemSynchronizer.getSonarLintJsonFiles(rootProject).stream()
-        .forEach(slFile -> files.add(FileSystemSynchronizer.toSubProjectFileDto(project, FileSystemSynchronizer.toFileDto(slFile, null))));
+        .forEach(slFile -> {
+          var dto = FileSystemSynchronizer.toFileDto(slFile, null);
```

Comment:
Progress monitor is not needed there?

Reply:
No, this is running "too fast" we don't want to spawn a progress monitor in the IDE, that would be overkill. If it moves by a pixel or not, no one cares when it takes 6 seconds.